app/renderer/playerView.js

```diff
@@ -107,41 +107,59 @@ function addTextSection(text)
     // Split the line if we have a >>
     var splitOnCarrotTags = text.split(">>");
     var nonTagText = splitOnCarrotTags.shift();
+
     // Reconstitute the rest of the line that has tags
     var tagText = "";
     if (splitOnCarrotTags.length > 0) {
         tagText = ">>" + splitOnCarrotTags.join(">>");
     }
 
     // If all we have left is a character name, style that like a tag
+
     if (/[A-Z]+:\s*$/.test(nonTagText)) {
         tagText = nonTagText + tagText;
         nonTagText = "";
     }
 
-    var splitIntoSpans = nonTagText.split(" ");
+    var arrayOfWords = nonTagText.split(" ");
+    var dialogueRegex = /^[A-Z]+:/
+    var isDialogueLine = dialogueRegex.test(arrayOfWords[0]);
 
-    var plainSpans = new Array();
-    var highlightedSpans = new Array();
+    var plainSpans = [];
+    var highlightedSpans = [];
 
     if (settings.getSync("enforceCharCounts")) {
         // ROBIN: Enforce character counts on this <p>.
         let charCount = 0;
-        let charCutoff = settings.getSync("charCountDanger");
-        for (let span of splitIntoSpans) {
-            charCount += span.length;
-            if (charCount < charCutoff) {
-                plainSpans.push(span);
-            } else {
-                highlightedSpans.push(span);
-            }
+        const MAX_NARRATION_COUNT = settings.getSync("narrativeCountDanger");
+        const MAX_DIALOGUE_COUNT = settings.getSync("dialogCountDanger");
+        let dangerousWordCount = isDialogueLine ? MAX_DIALOGUE_COUNT : MAX_NARRATION_COUNT;
+
+        for (var i = 0; i < arrayOfWords.length; i++) {
+          var currentWord = arrayOfWords[i];
+          charCount += currentWord.length;
+          if (charCount < dangerousWordCount) {
+              plainSpans.push(currentWord);
```

Comment:
We're using two spaces for indentation on this project— should be something you can set your editor to do automatically for you!